ha_dbs.md

```diff
@@ -0,0 +1,85 @@
+# Set up HA in the 3scale DBs
+
+As mentioned in [this other section](ha_3scale.md) of this repo, in order to set
+up HA for the 3scale DBs, we need to tell the operator that we are going to
+manage them ourselves. This applies to both the Backend and the System Redis,
+and also, the relational databases used by System, and starting from v2.10, also
+Zync.
+
+We can choose between different options when setting up the DBs. One of the
+first questions that we need to answer is whether we want to deploy them on
+premises or if we'd like to use a managed service provided by a cloud provider.
+Some users might be required by law to have everything on-premises. For users
+who are not, and are already using services provided by Amazon, for example,
+using their services also for the 3scale DBs might be the most convenient
+option.
+
+
+## Backend Redis and System Redis
+
+Redis offers replication. You need to set up multiple instances of Redis. One of
+them will be the leader and the others the replicas. Writes go to the leader and
+are propagated to the replicas. When the leader fails, a replica becomes the new
+leader. When the old leader comes back online it will become a replica of the
+new leader.
+
+In Redis, the replication mechanism is asynchronous. This means that when
+something is written to the leader, Redis answers to the client without waiting
+for the writes to be effective in the replicas. This is fast, but when there is
+a failover, we'll lose the data that has not been replicated yet.
+
+In the case of Backend, that trade-off makes sense. In the event of a failover,
+Backend might lose some reports. This means that some rate-limits could not be
+accurate and we could let more calls than configured for a brief period of time.
+Also, some statistics might not be exact. This is a trade-off that makes sense
+in the case of Backend. It needs to be fast because it is called on every API
+call (unless there's some caching policy enabled in APIcast), and failovers
+should be pretty rare, plus the amount of data not in the replicas should be
+pretty low. Apart from that, the Backend DB contains some information
+synchronized from System that could also be lost, but that can always be
+recovered by executing a rake task from System.
+
+Here are some options available to use as the Backend and System Redis:
+
+- [Redis with sentinels](https://redis.io/topics/sentinel). This is the option
+you need to choose if you want to deploy everything on-premises.
+- [AWS
+Elasticache](https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/WhatIs.html).
+Check the
+[docs](https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/Replication.Redis.Groups.html).
+It's [multi
+AZ](https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/AutoFailover.html),
+but not multi region.
```

Comment:
There is a new Elasticache called [Global Store](https://aws.amazon.com/elasticache/redis/global-datastore/?nc=sn&loc=3&dn=1) where it seems you can have multi region. 

You have have you active EC redis (with 1 active master and N slaves) on a region, and replicate the architecture on a new region for DR purpose (so, passive master, and N slaves). In case of first whole region down, you can promote second region to be primary (so passive master becomes active master...). More techical details at https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/Redis-Global-Datastore.html

Reply:
Thanks. I wasn't aware of this. I added it to the doc.

Reply:
nice!